pkg/observability/tracing/tracing.go

```diff
@@ -56,7 +46,7 @@ func Setup(ctx context.Context, resource *resource.Resource, protocol string) (_
 func registerTraceExporter(res *resource.Resource, exporter sdktrace.SpanExporter) (func(context.Context) error, error) {
 	options := []sdktrace.TracerProviderOption{
 		sdktrace.WithBatcher(exporter),
-		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithSampler(sdktrace.TraceIDRatioBased(defaultSamplingRatio)),
```

Comment:
Is there any way for the user to adjust this sampling rate?

Reply:
Good catch! I fixed it, so now `OTEL_TRACES_SAMPLER` and `OTEL_TRACES_SAMPLER_ARG` can be set together, so the sampler is configured dynamically, e.g.:

```sh
OTEL_TRACES_SAMPLER=traceidratio
OTEL_TRACES_SAMPLER_ARG=0.001
```

Reply:
(Defaults to [`parentbased_always_on`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#:~:text=Propagator%20only%20once.-,OTEL_TRACES_SAMPLER,-Sampler%20to%20be).

Reply:
Sounds good!